src/main/java/org/folio/rest/impl/ModTenantAPI.java

```diff
@@ -1,142 +1,157 @@
 import io.vertx.core.logging.Logger;
 import io.vertx.core.logging.LoggerFactory;
-import me.escoffier.vertx.completablefuture.VertxCompletableFuture;
 
 public class ModTenantAPI extends TenantAPI {
   private final Logger logger = LoggerFactory.getLogger(ModTenantAPI.class);
 
-  private static final String SHOULD_NOT_UPDATE_PROPS = "shouldUpdateProps";
   private static final String X_OKAPI_URL = "x-okapi-url";
   private static final String X_OKAPI_TENANT = "x-okapi-tenant";
-  private static final String MOD_CONFIGURATION_ENTRIES_URI = "/configurations/entries";
-  private static final String CONFIG_NAME = "configName";
-  private static final String ENABLED = "enabled";
+  private static final String X_OKAPI_TOKEN = "x-okapi-token";
   private static final String CONFIG_DIR_PATH = "config";
+  private static final String QUERY = "module==OAIPMH and configName==%s";
+  private static final String MODULE_NAME = "OAIPMH";
   private static final int CONFIG_JSON_BODY = 0;
   private ConfigurationHelper configurationHelper = ConfigurationHelper.getInstance();
 
   @Override
   public void postTenant(final TenantAttributes entity, final Map<String, String> headers,
       final Handler<AsyncResult<Response>> handlers, final Context context) {
-    loadConfigData(headers, context).thenAccept(response -> handlers.handle(Future.succeededFuture(response)))
-      .exceptionally(handleError(handlers));
+    loadConfigData(headers, handlers);
   }
 
-  private CompletableFuture<Response> loadConfigData(Map<String, String> headers, Context context) {
-    VertxCompletableFuture<Response> future = new VertxCompletableFuture<>(context);
-    List<String> configsSet = Arrays.asList("behavior","general","technical");
+  private void loadConfigData(Map<String, String> headers, Handler<AsyncResult<Response>> handlers) {
+    List<String> configsSet = Arrays.asList("behavior", "general", "technical");
+
+    List<Future> futures = new ArrayList<>();
+
+    configsSet.forEach(configName -> futures.add(processConfigurationByConfigName(configName, headers)));
+    CompositeFuture.all(futures)
+      .onComplete(future -> {
+        String message;
+        if (future.succeeded()) {
+          message = "Configurations has been set up successfully";
+        } else {
+          message = future.cause().getMessage();
+        }
+        handlers.handle(Future.succeededFuture(buildResponse(message)));
+      });
+  }
 
+  private Future processConfigurationByConfigName(String configName, Map<String, String> headers) {
+    Promise promise = Promise.promise();
     String okapiUrl = headers.get(X_OKAPI_URL);
     String tenant = headers.get(X_OKAPI_TENANT);
+    String token = headers.get(X_OKAPI_TOKEN);
 
-    HttpClientInterface httpClient = HttpClientFactory.getHttpClient(okapiUrl, tenant, true);
-    List<CompletableFuture<Void>> completableFutures = new ArrayList<>();
+    ConfigurationsClient client = new ConfigurationsClient(okapiUrl, tenant, token);
```

Comment:
The client is created 3 times. Can you create it once and pass it to this method?